File: test/CompareIntegersTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class CompareIntegersTest {

    @Test
    void testEqualIntegers() {
        int result = CompareIntegers.compareIntegers(10, 10);
        assertEquals(1, result, "Expected result to be 1 when the integers are equal.");
    }

    @Test
    void testFirstIntegerGreater() {
        int result = CompareIntegers.compareIntegers(20, 10);
        assertEquals(2, result, "Expected result to be 2 when the first integer is greater.");
    }

    @Test
    void testSecondIntegerGreater() {
        int result = CompareIntegers.compareIntegers(5, 15);
        assertEquals(3, result, "Expected result to be 3 when the second integer is greater.");
    }

    @Test
    void testNegativeIntegers() {
        int result = CompareIntegers.compareIntegers(-5, -10);
        assertEquals(2, result, "Expected result to be 2 when the first integer is greater, even if negative.");
    }

    @Test
    void testMixedSignIntegers() {
        int result = CompareIntegers.compareIntegers(-5, 10);
        assertEquals(3, result, "Expected result to be 3 when the second integer is greater with mixed signs.");
    }
}

File: python_tests/compare_integers_test.py
import unittest
class compare_integers_test(unittest.TestCase):
    def test_equal(self):
        """Check when both integers are equal."""
        self.assertEqual(compare_integers(10, 10), 1)  # a == b

    def test_a_greater(self):
        """Check when the first integer is greater."""
        self.assertEqual(compare_integers(20, 10), 2)  # a > b

    def test_b_greater(self):
        """Check when the second integer is greater."""
        self.assertEqual(compare_integers(5, 15), 3)  # b > a